rmw/include/rmw/qos_profiles.h

```diff
@@ -180,6 +180,103 @@ rmw_qos_profile_check_compatible(
   char * reason,
   size_t reason_size);
 
+/// Get compatible QoS policies for a subscription.
+/**
+ * Given one or more publisher QoS profiles, return a QoS profile for a subscirption
+ * that is compatible with the majority of the publisher profiles while maintaining the highest
+ * level of service possible.
+ *
+ * This function modifies the provided subscription QoS profile in order to make it compatible
+ * with the publisher profiles.
+ * It should only modify those members of the QoS profile that it needs to improve compatibility.
+ * For example, for DDS implementations this function should not modify history depth, which
+ * has no role in determining compatibility.
+ *
+ * It's possible not all input QoS profiles are compatible with the output subscription QoS
+ * profile.
+ * You can optionally provide a boolean array that will by set by this function to indicate
+ * which publisher QoS profiles are compatible with the result.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | Yes
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[in] publisher_profiles: An array of QoS profiles for publishers.
+ * \param[in] publisher_profiles_length: The number of elements in `publisher_profiles`.
+ * \param[out] subscription_profile: QoS policies that are compatible with the majorty of
```

Comment:
```suggestion
 * \param[out] subscription_profile: QoS policies that are compatible with the majority of
```
A side from the typo, could it be that "majority" is somehow ambiguous here? I mean, we probably want to know if the policy is "compatible enough" to be used and that implies that is compatible with all input publisher profiles. Is my understanding correct?

Reply:
For DDS implementations, it will be compatible with all input publisher profiles, but I'm purposefully using the word "majority" here to leave room for non-DDS implementations that may not be able to match all input publishers (their QoS compatibility rules may be different).